### PR TITLE
Fixing client code - passing secret provider

### PR DESCRIPTION
**Pull request by GunaKKIBM**

### CODE REVIEW

File: pkg/ibmcloudprovider/volume_provider.go [hunk 1]
```diff
@@ -117,7 +103,7 @@ func NewIBMCloudStorageProvider(configPath string, clusterVolumeLabel string, lo
 		ProviderName:   providerName,
 		ProviderConfig: conf,
 		Registry:       registry,
-		ClusterInfo:    clusterInfo,
```
Comment (GunaKKIBM) [inline pkg/ibmcloudprovider/volume_provider.go:120]: Add comments

**Review by ambiknai — COMMENTED** [review]
@GunaKKIBM  Will this have any impact on our sanity tests

File: pkg/ibmcloudprovider/ibm_volume_provider.go [hunk 2]
```diff
@@ -29,5 +28,5 @@ import (
 type CloudProviderInterface interface {
 	GetProviderSession(ctx context.Context, logger *zap.Logger) (provider.Session, error)
 	GetConfig() *config.Config
-	GetClusterInfo() *utils.ClusterInfo
```
Comment (sameshai) [inline pkg/ibmcloudprovider/ibm_volume_provider.go:32]: Did we confirm if this is being used by any other library. This is common library.

Comment (GunaKKIBM) [reply pkg/ibmcloudprovider/ibm_volume_provider.go:32]: Yes, Sameer checked this. This interface is used only by block csi driver. A similar interface is declared for file csi driver in it's own library.

**Review by sameshai — APPROVED** [review]

File: pkg/watcher/pv_watcher_test.go [hunk 3]
```diff
@@ -73,10 +78,20 @@ func TestAddTags(t *testing.T) {
...
+	broadcaster.StartLogging(glog.Infof)
+	clientset := fake.NewSimpleClientset()
+	eventInterface := clientset.CoreV1().Events("")
+	broadcaster.StartRecordingToSink(&v1core.EventSinkImpl{Interface: eventInterface})
+
 	pvw := &PVWatcher{
 		provisionerName: "ibm-csi-driver",
 		logger:          logger,
 		config:          conf,
+		cloudProvider:   fakeIBMCloudStorageProvider,
```
Comment (GunaKKIBM) [inline pkg/watcher/pv_watcher_test.go:93]: Check this

Comment (GunaKKIBM) [reply pkg/watcher/pv_watcher_test.go:93]: added fake in test cases

**Review by ambiknai — APPROVED** [review]

**Pull request merged**